### ISSUE TITLE
'fisher update' hangs.

### ISSUE THREAD

**Issue by RafaelCosman** [fisherman/fisherman#12]
On C9 Ubuntu machine.

**Comment by bucaran** [comment]
Yes, this is a problem in machines other than OS X and ubuntu/trusty64, the only two platforms I have tested this well. The problem has to do with my use of `find` it seems. The milestone I am working towards will fix this by adding a `--cache` to `fisher` to retrieve the cache plain contents.

https://github.com/fisherman/fisherman/issues/11

**Comment by RafaelCosman** [comment]
Thanks for working on this, @bucaran.

**Comment by simnalamburt** [comment]
:+1:

**Comment by bucaran** [comment]
@RafaelCosman @simnalamburt 

Guys, I was mistaken and replied to this message shortly after another user had reported a similar issue and claimed it had to do with `find(1)`. This in fact, has nothing to do with that, (probably), and it's just the default behavior of `update`.

Fisherman has strong ties to the UNIX philosophy, at least it says so in the manual :smile: and one of the things that are common across commands is that streams are prioritized. 

So, `fisher update` will hang because is waiting for input. This allows cool things like:

```fish
fisher --cache | fisher udpate
```

or

```fish
fisher update < bundle
```

**Comment by rif** [comment]
So what is the recommend way to list/update all installed plugins?

**Comment by bucaran** [comment]
Hey @RafaelCosman 

Just, `fisher update plugin1 plugin2 ...`.

```fish
fisher update -h
usage: fisher update [<name or url> ...] [--me] [--quiet] [--help]

       -m --me  Update Fisherman
    -q --quiet  Enable quiet mode
     -h --help  Show usage help
```

If you want to update all the plugins:

```
fisher --cache | fisher update
```

If you want to update Fisherman itself:

```
fisher update --me
```

**Comment by simnalamburt** [comment]
I think making `fisher update` to update all plugins is the best default. Most package managers like `npm`, `gem`, etc update all installed plugin when executed with `~~ update` and I think most users are used to it.

Yes `fisher --cache | fisher update` fits very well with UNIX philosophy but `fisher update` is also good default, and I think both can coexist with [`isatty(3)`][isatty] just like as the `python` binary does.

How do you think?

[isatty]: http://linux.die.net/man/3/isatty

**Comment by bucaran** [comment]
@simnalamburt Alright. Why not?  :+1:

**Comment by bucaran** [comment]
Fixed in 0.3.0

**Comment by simnalamburt** [comment]
Perfect
